pkgs/development/haskell-modules/configuration-common.nix

```
@@ -612,6 +612,24 @@ self: super: {
     '';
   });
 
+  d-bus = let
+    # The latest stack release is missing necessary patches for recent compilers
+    # https://github.com/Philonous/d-bus/issues/24
+    newer = overrideSrc super.d-bus {
```

Please mark that this version is picked from git.

```suggestion
    newer = overrideSrc super.d-bus {
      version = "unstable-2021-01-08";
```